crates/curp/src/server/raw_curp/log.rs

```diff
@@ -175,15 +217,32 @@ impl<C: Command> LogEntryVecDeque<C> {
 
     /// check whether the log entry range [li,..) exceeds the batch limit or not
     fn has_next_batch(&self, left: usize) -> bool {
-        if let (Some(&cur_size), Some(&last_size)) =
-            (self.batch_index.get(left), self.batch_index.back())
-        {
-            let target_size = cur_size.overflow_add(self.batch_limit);
-            target_size <= last_size
+        if let Some(&offset) = self.batch_index.get(left) {
+            offset != 0
+                || (self.first_entry_at_last_batch == left
+                    && self.batch_limit == self.last_batch_size)
+                || left == self.batch_index.len() - 1
```

Comment:
Why does the condition `left == self.batch_index.len() - 1` indicates that the log has next batch?

Reply:
After the modification above, this part of code has been refactored.